notification/src/main/java/org/apache/atlas/notification/spool/Spooler.java

```diff
@@ -100,23 +102,27 @@ boolean write(List<String> messages) {
         return ret;
     }
 
-    private void writeToFailedMessages(List<String> messages) {
+    private void writeToFailedMessages(List<KeyValue<String,String>> messages) {
         if (failedMessagesLogger != null) {
-            for (String message : messages) {
-                failedMessagesLogger.log(message);
+            for (KeyValue<String,String> message : messages) {
+                //Bellemare TODO - Does this affect a test / formatting?
+                failedMessagesLogger.log(message.toString());
             }
         }
     }
 
-    private boolean writeInternal(List<String> messages) {
+    private boolean writeInternal(List<KeyValue<String,String>> messages) {
         boolean ret = false;
 
         try {
             byte[]     lineSeparatorBytes = SpoolUtils.getLineSeparator().getBytes(SpoolUtils.DEFAULT_CHAR_SET);
             DataOutput pw                 = indexManagement.getSpoolWriter();
 
-            for (String message : messages) {
-                pw.write(message.getBytes(SpoolUtils.DEFAULT_CHAR_SET));
+            for (KeyValue<String,String> message : messages) {
+                //TODO - Bellemare - Serialize this with a proper format
```

Comment:
@mneethiraj Ah yes, I think this would be an area where simply placing the key in HookNotification may make life a lot easier.